src/msg/async/AsyncConnection.cc

```diff
@@ -552,6 +552,22 @@ int AsyncConnection::send_message(Message *m)
     m->put();
     return 0;
   }
+  auto& conf = msgr->cct->_conf;
+  auto drop_mon_forward_msgs = conf.get_val<double>("ms_inject_drop_mon_forward_msgs");
```

Comment:
This will be incredibly expensive. Looking up configs like this for every message send is not going to work.

This change only affects the mon and it should go in some centralized message send method there.

Reply:
@batrick moved to the monitor code and save few checks